### PR TITLE
BE-89 Refactor character ranges on document detail and document diff …

### PR DESCRIPTION
**Pull request by DeMensa123**
…endpoints

### CODE REVIEW

File: antiplag/models.py [hunk 1]
```diff
@@ -38,6 +38,107 @@ def __str__(self):
...
+        prev = x['intervals'][0]
+        count += x['intervals'][1]
+
+        if x['intervals'][1] == 1:
+            docs.append(x['doc'])
+
+        else:
+            del docs[docs.index(x['doc'])]
+
+    return intervals
```
Comment (OliverKanat) [inline antiplag/models.py:139]: I would move it to `utils.py` module or elsewhere.

File: antiplag/serializers.py [hunk 2]
```diff
@@ -61,7 +61,8 @@ class Meta:
 
 class DocumentResultSerializer(serializers.ModelSerializer):
     result = ResultSerializer(read_only=True)
+    intervals = serializers.ReadOnlyField(source="result.intervals")
 
     class Meta:
         model = Document
-        fields = ("name", "result", "text")
+        fields = ("name", "result", "text", "intervals")
```
Comment (OliverKanat) [inline antiplag/serializers.py:68]: Are intervals needed for every endpoint that uses this serializer?

**Review by OliverKanat — CHANGES_REQUESTED** [review]

**Comment by OliverKanat** [comment]
Tests are missing :(

File: antiplag/serializers.py (hunk 2 above)
Comment (DeMensa123) [reply antiplag/serializers.py:68]: Nop, DocumentResultSerializer is used only in DocumentDetail endpoint.

**Review by Hemii — APPROVED** [review]

File: antiplag/utils.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+        elif top["to"] <= x["to"]:
+            top["to"] = x["to"]
+            stack.pop()
+            stack.append(top)
+
+    starts = []
+    ends = []
+
+    for x in stack:
```
Comment (OliverKanat) [inline antiplag/utils.py:34]: What is `x`? Please rename all similar variables to meaningful name.

**Review by OliverKanat — CHANGES_REQUESTED** [review]

File: antiplag/tests/test_intervals.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
+from django.test import TestCase
+
+from ..models import Submission, Document, Result
+
+
+class ResultModelTestCase(TestCase):
```
Comment (OliverKanat) [inline antiplag/tests/test_intervals.py:6]: Please add some tests for edge cases. For example this case:

```
if len(indices) <= 1:
        return indices
```

File: antiplag/utils.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                "doc": {
+                    "id": item["id"],
+                    "name": item["name"],
+                    "percentage": item["percentage"],
+                },
+            }
+        )
+
+    size_1 = len(starts)
+    size_2 = len(ends)
```
Comment (OliverKanat) [inline antiplag/utils.py:54]: What is `size_1` or `size_2`? Name it correctly.

**Review by OliverKanat — CHANGES_REQUESTED** [review]

**Review by OliverKanat — APPROVED** [review]

**Review by Hemii — APPROVED** [review]
